src/Servers/IIS/AspNetCoreModuleV2/RequestHandlerLib/filewatcher.cpp

```diff
@@ -30,33 +36,25 @@ FILE_WATCHER::~FILE_WATCHER()
 
 void FILE_WATCHER::WaitForMonitor(DWORD dwRetryCounter)
 {
-    if (m_hChangeNotificationThread != NULL)
+    if (m_hChangeNotificationThread == nullptr)
     {
-        DWORD dwExitCode = STILL_ACTIVE;
+        return;
+    }
 
-        while (!m_fThreadExit && dwRetryCounter > 0)
+    while (!m_fThreadExit && dwRetryCounter-- > 0)
```

Comment:
What happens if the thread doesn't exit after the loop completes?

And, should we log in that case?

Reply:
Good idea, done.

Reply:
One option if we want to de-risk this in .NET 8 is to enable the termination behavior behind a flag.

Reply:
What is the thread doing that could be "bad" if it's still running?

Reply:
Offline discussion: The thread is holding onto state that can have undefined behavior if accessed when the class is destructed. e.g. `FILE_WATCHER*` and the `m_pShutdownEvent` and `m_hCompletionPort` events.

Which brings up another thing, those events are never closed via `CloseHandle` it looks like.